### ISSUE TITLE
Keyboard support for AutocompleteView (and SearchTextView)

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#15069]
## 📝 Provide a description of the improvement

ATM the component shares keyboard support with `SearchTextView`. It's far from perfect because you have to tab first to get access to the list and to browse it. It's unlike any autocomplete out there in the industry. 

The component should have a persistent selection in the text field and up/down arrows selecting (but no focusing) filtered view's items and updating the value of the input. Check out your browser's URL input to get the idea.

This will require:
* Additional stuff in the `FilteredView` interface (for selecting but not turning on or focusing items).
* Keyboard interaction logic
* Undoing some existing focus cycling and tracking logic (I suppose filtered view's items should no longer be focusable).

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.
